### ISSUE TITLE
Public Headers

### ISSUE THREAD

**Issue by julianxhokaxhiu** [liblava/liblava#3]
Hi,

I've successfully managed to build the library under 32bit platform thanks to #2.

Now although I'm facing another problem. As I'd like to use this library as a pre-built library, I'm missing the include headers. Is there a possibility to have public headers installed via CMake when running the install target?

Would be fine to have them generated as well, one per target.

Thank you in advance,
Julian

**Comment by TheLavaBlock** [comment]
Hi Julian, Good point. 

Once **C++20 modules** are supported in CMake, **liblava** will be upgraded completely.
This means that header files are no longer needed.

Currently it has to be done manually or via script. Unfortunately.

**Comment by julianxhokaxhiu** [comment]
Hi @TheLavaBlock,

I am currently checking the library again for some progress, and I notice this issue is still open. As C++ 20 modules on CMake will not get supported until who known when, is there a way to workaround this issue? You mentioned then it can be done manually or via a script. I would be fine with both, but can you please give me a hint on what can be done?

I was checking this CMake function, but I'm not sure it's exactly what we need: https://cmake.org/cmake/help/v3.17/module/GenerateExportHeader.html

Any help on this topic is appreciated :)

Thank you in advance,
Julian

**Comment by pezcode** [comment]
The canonic way to do it would be with the `PUBLIC_HEADERS` target property + specifying an install rule for each target: https://stackoverflow.com/a/40285224/862300

A really easy alternative is something like
```
install(DIRECTORY liblava/ DESTINATION include/liblava FILES_MATCHING PATTERN "*.hpp")
```

**Comment by TheLavaBlock** [comment]
I like the easy alternative.

**Comment by TheLavaBlock** [comment]
And yeah, C++20 modules with CMake will take a while.

**Comment by julianxhokaxhiu** [comment]
But `*.hpp` files to contain code. So now I think probably another way of including liblava is just adding the whole `/liblava` dir into the include path, there is no need to pre-compile it.

Although at the same time, pre-compiling it would provide faster compilation for whichever library may link to it. So I'm not sure we can merge the pre-built lib files with public headers, which do contain code anyway inside. Or am I missing something here?

**Comment by pezcode** [comment]
There's no code in .hpp files, it's just a regular C++ header. But yes, you'd still need an `install` directive for the libraries.

I can give this a try tomorrow (no promises 🤡)

**Comment by pezcode** [comment]
@julianxhokaxhiu PR with preliminary install support: https://github.com/liblava/liblava/pull/18

Feel free to give it a try and report back

**Comment by TheLavaBlock** [comment]
@julianxhokaxhiu can this be closed? Think #18 is doing a good job.

**Comment by julianxhokaxhiu** [comment]
Absolutely! Thank you again
